lib/Controller/MailboxesController.php

```diff
@@ -147,20 +147,26 @@ public function patch(int $id,
 	 *
 	 * @param bool $init
 	 * @param string|null $query
+	 * @param string|'all' $criteria
```

Comment:
I understand.

But having 3 booleans, would make the route very long:

```diff
diff --git a/appinfo/routes.php b/appinfo/routes.php
index 4f669ad09..068db5dea 100644
--- a/appinfo/routes.php
+++ b/appinfo/routes.php
@@ -91,8 +91,11 @@ return [
                ],
                [
                        'name' => 'mailboxes#sync',
-                       'url' => '/api/mailboxes/{id}/sync',
-                       'verb' => 'POST'
+                       'url' => '/api/mailboxes/{id}/sync/{criteria}',
+                       'verb' => 'POST',
+                       'defaults' => [
+                               'criteria' => 'all'
+                       ]
                ],
                [
                        'name' => 'mailboxes#clearCache',
```

We could maybe use an int that just copies Horde's implementation `Horde_Imap_Client::SYNC_NEWMSGSUIDS | Horde_Imap_Client::SYNC_FLAGSUIDS | Horde_Imap_Client::SYNC_VANISHEDUIDS`?

Reply:
I wouldn't include the booleans in the route but send them as GET parameters like `.../sync?syncNew=true`

Reply:
Can we do this? If you're busy I can take over. Just let me know :pray:

Reply:
right now I'm busy, playing with attachments. So, feel free to take this PR over if you like. 

Otherwise, I'll probably be able to go back at it in the next coming days.